### PR TITLE
Finish validating rfq against provided offering

### PR DESCRIPTION
**Pull request by kirahsapong**
This PR addresses #94

### CODE REVIEW

File: packages/protocol/tests/rfq.spec.ts [hunk 1]
```diff
@@ -158,6 +158,156 @@ describe('Rfq', () => {
...
+      }
+    })
+    it('throws an error if payinMethod kind cannot be validated against the provided offering\'s payinMethod kinds', async () => {
+      const rfq = Rfq.create({
+        ...rfqData,
+        data: {
+          ...rfqData.data,
+          payinMethod: {
+            ...rfqData.data.payinMethod,
+            kind: 'POKEMON'
```
Comment (jiyoontbd) [inline packages/protocol/tests/rfq.spec.ts:240]: 😃

**Review by jiyoontbd — APPROVED** [review]

File: packages/protocol/tests/rfq.spec.ts [hunk 2]
```diff
@@ -158,6 +158,156 @@ describe('Rfq', () => {
     })
   })
 
+  describe('verifyOfferingRequirements', () => {
+    let offering
+    let rfqData
```
Comment (KendallWeihe) [inline packages/protocol/tests/rfq.spec.ts:163]:
```suggestion
    let offering: Offering
    let rfqData: RfqData
```

Comment (KendallWeihe) [reply packages/protocol/tests/rfq.spec.ts:163]: wait, might need to be nullable too, whatevs not a big deal

Comment (KendallWeihe) [reply packages/protocol/tests/rfq.spec.ts:163]: actually, could you reuse what's in `rfq.spec.ts`? https://github.com/TBD54566975/tbdex-js/blob/e31fa7ba252e67574897fab180246023c6920a08/packages/protocol/tests/rfq.spec.ts#L7

oh, now I see it's of type `Rfq` not `RfqData` (variable name should just be `rfq` then)

Comment (KendallWeihe) [reply packages/protocol/tests/rfq.spec.ts:163]: (deleted a comment b/c I spoke too soon)

File: packages/protocol/tests/rfq.spec.ts [hunk 3]
```diff
@@ -158,6 +158,156 @@ describe('Rfq', () => {
...
+    before(() => {
+      async function initMocks() {
```
Comment (KendallWeihe) [inline packages/protocol/tests/rfq.spec.ts:165]: I think you can make the entire `before` function `async` and then you don't need a `initMocks()` function block, everything could just go within the `before()` function block, like this...

```typescript
before(async () => {
```

**Review by KendallWeihe — APPROVED** [review]
lgtm, though the doc commenter is complaining about missing member variables?

Screenshot 2023-12-19 at 10 56 15 AM

**Pull request merged**